hotrod/src/test/java/io/hyperfoil/hotrod/HotRodTest.java

```diff
@@ -0,0 +1,68 @@
+package io.hyperfoil.hotrod;
+
+import java.util.Map;
+import java.util.Properties;
+
+import org.infinispan.client.hotrod.test.HotRodClientTestingUtil;
+import org.infinispan.commons.test.TestResourceTracker;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.global.GlobalConfigurationBuilder;
+import org.infinispan.manager.DefaultCacheManager;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.server.core.admin.embeddedserver.EmbeddedServerAdminOperationHandler;
+import org.infinispan.server.hotrod.HotRodServer;
+import org.infinispan.server.hotrod.configuration.HotRodServerConfigurationBuilder;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import io.hyperfoil.api.config.Benchmark;
+import io.hyperfoil.api.statistics.StatisticsSnapshot;
+import io.hyperfoil.core.session.BaseScenarioTest;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+
+@RunWith(VertxUnitRunner.class)
+public class HotRodTest extends BaseScenarioTest {
+
+   protected HotRodServer hotrodServers;
+
+   @Before
+   public void before(TestContext ctx) {
+      super.before(ctx);
+      TestResourceTracker.setThreadTestName("hyperfoil-HotRodTest");
+
+      GlobalConfigurationBuilder globalBuilder = new GlobalConfigurationBuilder();
+      globalBuilder.globalState().persistentLocation("/tmp");
+      globalBuilder.globalState().enabled(true);
+      globalBuilder.security().authorization().disable();
+
+      HotRodServerConfigurationBuilder serverBuilder = new HotRodServerConfigurationBuilder();
+      serverBuilder.adminOperationsHandler(new EmbeddedServerAdminOperationHandler());
+
+      ConfigurationBuilder cacheBuilder = new ConfigurationBuilder();
+
+      EmbeddedCacheManager em = new DefaultCacheManager(globalBuilder.build());
+      em.createCache("my-cache", cacheBuilder.build());
+      hotrodServers = HotRodClientTestingUtil.startHotRodServer(em, serverBuilder);
+   }
+
+   @After
+   public void after(TestContext ctx) {
+      super.after(ctx);
+      if (hotrodServers != null) {
+         hotrodServers.stop();
+      }
+   }
+
+   @Test
+   public void test() {
+      Properties properties = new Properties();
+      properties.put("hotRodPort", hotrodServers.getPort());
+      Benchmark benchmark = loadScenario("scenarios/HotRodTest.hf.yaml", properties);
```

Comment:
I've already mentioned that properties replacement is not acceptable. Please use override the loadScenario(...) in the test to change the port in the BenchmarkBuilder rather than replacing properties.

Reply:
Done